src/js/jsx/sections/libraries/assets/AssetPreviewImage.jsx

```diff
@@ -49,24 +49,54 @@ define(function (require, exports, module) {
         },
         
         componentWillMount: function () {
-            // On mount, get the rendition of this element
-            var element = this.props.element;
+            this._updateRenditionPath();
+        },
+        
+        componentWillUpdate: function () {
+            this._updateRenditionPath();
+        },
+        
+        shouldComponentUpdate: function (nextProps, nextState) {
+            return this._lastModified !== nextProps.element.modified ||
```

Comment:
`_lastModified` is not declared anywhere, can you declare it with a JSDoc at the top of the component?